watertap/tools/analysis_tools/loop_tool/tests/test_loop_tool.py

```diff
@@ -0,0 +1,332 @@
+    get_mpi_comm_process,
+)
+
+__author__ = "Alexander V. Dudchenko (SLAC)"
+
+_this_file_path = os.path.dirname(os.path.abspath(__file__))
+
+
+def diff_dict_check(dicta, dictb):
+    for key in dicta:
+        if key != "dir":
+            if isinstance(dicta[key], dict):
+                diff_dict_check(dicta[key], dictb[key])
+
+            elif dicta[key] != dictb[key]:
+                return False
+    return True
+
+
+@pytest.fixture()
+def loop_test_options_setup():
+    lp = loopTool(
+        _this_file_path + "/test_all_options.yaml",
+        build_function=ro_setup.ro_build,
+        initialize_function=ro_setup.ro_init,
+        optimize_function=ro_setup.ro_solve,
+        saving_dir=_this_file_path,
+        save_name="ro_with_erd",
+        execute_simulations=False,
+        number_of_subprocesses=1,
+    )
+    lp.build_run_dict()
+    """ used to generate test file"""
+    # with open("test_expected_option_directory.yaml", "w") as file:
+    #     documents = yaml.dump(lp.sweep_directory, file)
+    if has_mpi_peer_processes() == False or (
+        has_mpi_peer_processes() and get_mpi_comm_process().Get_rank() == 0
+    ):
+        with open(
+            _this_file_path + "/test_expected_option_directory.yaml", "r"
+        ) as infile:
+            expected_run_dict = yaml.safe_load(infile)
+    else:
+        expected_run_dict = None
+    return lp, expected_run_dict
+
+
+@pytest.fixture()
+def loop_sweep_setup():
+    lp = loopTool(
+        _this_file_path + "/test_sweep.yaml",
+        build_function=ro_setup.ro_build,
+        initialize_function=ro_setup.ro_init,
+        optimize_function=ro_setup.ro_solve,
+        saving_dir=_this_file_path,
+        save_name="ro_with_erd",
+        execute_simulations=False,
+        number_of_subprocesses=1,
+    )
+    lp.build_run_dict()
+    """ used to generate test file"""
+    # with open("test_expected_sweep_directory.yaml", "w") as file:
+    #     documents = yaml.dump(lp.sweep_directory, file)
+    if has_mpi_peer_processes() == False or (
+        has_mpi_peer_processes() and get_mpi_comm_process().Get_rank() == 0
+    ):
+        with open(
+            _this_file_path + "/test_expected_sweep_directory.yaml", "r"
+        ) as infile:
+            expected_run_dict = yaml.safe_load(infile)
+    else:
+        expected_run_dict = None
+    return lp, expected_run_dict
+
+
+@pytest.fixture()
+def loop_diff_setup():
+    lp = loopTool(
+        _this_file_path + "/test_diff.yaml",
+        build_function=ro_setup.ro_build,
+        initialize_function=ro_setup.ro_init,
+        optimize_function=ro_setup.ro_solve,
+        saving_dir=_this_file_path,
+        save_name="ro_with_erd",
+        execute_simulations=False,
+        number_of_subprocesses=1,
+    )
+    lp.build_run_dict()
+    """ used to generate test file"""
+    # with open("test_expected_diff_directory.yaml", "w") as file:
+    #     documents = yaml.dump(lp.sweep_directory, file)
```

Comment:
```suggestion
```

Reply:
I want to leave these in, so its easy to recreate the test file in the future. (You can't really create it manually)...